omod/src/main/java/org/openmrs/module/owa/web/controller/AddAppController.java

```diff
@@ -95,8 +98,9 @@ public String upload(@RequestParam("file") MultipartFile file, HttpServletReques
             
         } else {
             message = messageSourceService.getMessage("owa.blank_zip");
-            log.warn("App is not a zip archive");
-            return message;
+            log.warn("Zip file is empty");
+            model.addAttribute("message", message);
```

Comment:
@SandeepRaparthi1 Do not use model to display messages. We should use MessageSourceService to display messages. Sending you email in detail.